### PR TITLE
revert: use eplison to handle all zero cosine values (#1696)

### PR DESCRIPTION
**Pull request by eddyxu**
This reverts commit a178ae512f81a7eff858f7ca80a974ecc0a6c01d.

### CODE REVIEW

**Review by wjones127 — APPROVED** [review]

File: rust/lance-linalg/src/distance/cosine.rs [hunk 1]
```diff
@@ -211,7 +211,7 @@ where
     let y_sq = dot(y, y);
     let xy = dot(x, y);
     // 1 - xy / (sqrt(x_sq) * sqrt(y_sq))
-    1.0 - xy / (x_norm * y_sq.sqrt() + f32::epsilon())
+    1.0 - xy / x_norm * y_sq.sqrt()
```
Comment (rok) [inline rust/lance-linalg/src/distance/cosine.rs:214]:
```suggestion
    1.0 - xy / (x_norm * y_sq.sqrt())
```

File: rust/lance-linalg/src/distance/cosine.rs [hunk 2]
```diff
@@ -227,7 +227,7 @@ where
     let xy = dot(x, y);
     // 1 - xy / (sqrt(x_sq) * sqrt(y_sq))
     // use f64 for overflow protection.
-    1.0 - (xy / (x_norm * y_norm + f32::epsilon()))
+    1.0 - (xy / x_norm * y_norm)
```
Comment (rok) [inline rust/lance-linalg/src/distance/cosine.rs:230]:
```suggestion
    1.0 - (xy / (x_norm * y_norm))
```

Comment (eddyxu) [reply rust/lance-linalg/src/distance/cosine.rs:230]: Good catch!

Comment (wjones127) [reply rust/lance-linalg/src/distance/cosine.rs]: Make this a reminder that you should be adding many more tests to validate your math is correct :)

**Pull request merged**